### PR TITLE
Stats revamp v2 action cards

### LINKED ISSUE #16581: [android] Stats:  Action Cards

### PR DESCRIPTION
**Pull request by ravishanker**
This PR adds action cards to Insights view on Stats that appear when data meets certain criteria

Fixes #16581 
- [x] Depends on this [FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2434)

| Reminder and Schedule | Grow |
|---|---|
|Screenshot_20220521_224158| Screenshot_20220521_224232| 

To test:

Setup:

- Go to App Settings (Tap on Avatar at the top right hand corner on My Site -> Find App Settings)
- Select Debug Settings
- Find StatsRevampV2FeatureConfig under Features in development enable it and restart app
- Launch / Re-Launch app
- Go to Stats either using quick links or menu 
- Switch to Insights tab if necessary
- Make sure **Most Popular Time** and **Total Followers** cards are visible on Insights tab. If those cards are not present then add them through stats management using either [+ add new stats card ] at the bottom or clicking ⚙️ icon at the top right hand corner.

Test 1: 

- A **Reminder action card** appears below **Most Popular Time** card 
- Click on the SET REMINDERS, it should take to blogging reminders flow on my site.
- Go through the blogging reminders setup and make sure setup works
- NOTE: The guide only appears if best day and hour has numbers

Test 2:
- For this a draft post is required.  Choose a site with a draft post or create one and return to Stats
- A **Schedule action card**  appears below **Most Popular Time** card if there are any drafts locally and reminder card is not showing (unlike in the image above, they do not appear together)
- Click on the SET SCHEDULE, it should take to DRAFTS tab on Posts view.
- NOTE: The guide only appears if there's a draft 

Test 3:
- A **Grow action card** as appears below **Total Followers** card 
- Click on the CHECK THE COURSE, it should open Followers page on WordPress.com support site.
- NOTE: The guide only appears if total followers <= 1

Test 4:
- Click on DISMISS on any or all of the cards and the card/s should disappear.  They wouldn't appear again for the site

NOTE:
- At the moment, action cards only appear once and do not appear again once dismissed or action taken on the card.

## Regression Notes
1. Potential unintended areas of impact
Ensure cards appear and disappear without affecting current cards 

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and updated tests

3. What automated tests I added (or what prevented me from doing so)
Existing and updated tests

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by irfano** [comment]
- Test 1 worked. But after setting the reminder, **the same action card still stays active**.
- **I couldn't see other action cards** on test 2 and test 3 cases. Am I doing something wrong? I have a draft post, but I still can't see the schedule action card.
- Is this an action card, will we implement this too? 👇🏻 
image 1

File: WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/sections/insights/usecases/MostPopularInsightsUseCase.kt
Comment (irfano) [inline WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/sections/insights/usecases/MostPopularInsightsUseCase.kt]: np: Can we pass domainModel to `addActionCards()` we already had instead of calling `getMostPopularInsights()` again? `getMostPopularInsights` will cause an extra SQL call.

File: WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/sections/insights/usecases/TotalFollowersUseCase.kt
Comment (irfano) [inline WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/sections/insights/usecases/TotalFollowersUseCase.kt]: Function name is `add...` but it seems this function only `dismiss()` thing. Why do we dismiss in an add function?

**Review by irfano — COMMENTED** [review]
Just a few comments and questions. The code LGTM. I'll approve when I'll be able to test missing test cases.

File: WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/sections/insights/usecases/TotalFollowersUseCase.kt
Comment (ravishanker) [reply WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/sections/insights/usecases/TotalFollowersUseCase.kt]: Reworked this part now, it is supposed to add actually.  Thanks

**Comment by ravishanker** [comment]
> * Test 1 worked. But after setting the reminder, **the same action card still stays active**.

Yes, that was the question I asked on channel.  It is reworked now entirely

>     * **I couldn't see other action cards** on test 2 and test 3 cases. Am I doing something wrong? I have a draft post, but I still can't see the schedule action card.
This part is reworked as well. Please try again.  Thanks

>     * Is this an action card, will we implement this too? 👇🏻
>       image 1

That was in v1 iteration. There are only three action cards with dark background on final design!

**Comment by irfano** [comment]
- When "SET REMINDERS" button is clicked, we're removing the card and not showing it again. Should we check that the user really sets the reminder, or can just clicking "SET REMINDERS" button be enough to assume the action is taken?
- When a site is created, the owner of the site follows the site automatically. So, the follower count is not 0 unless the owner unfollow their site. Should we change the rule for guide and action cards, `count == 0` -> `count <= 1`?
- I can't see the grow action card. Is there any other required rule?


- Will we develop this card in v2?
Body

**Comment by ravishanker** [comment]
> When "SET REMINDERS" button is clicked, we're removing the card and not showing it again. Should we check that the user really sets the reminder, or can just clicking "SET REMINDERS" button be enough to assume the action is taken?

The idea is to just check if they would take action at all, not whether they would actually complete it or not.  That would probably increase the scope far beyond the feature

> * When a site is created, the owner of the site follows the site automatically. So, the follower count is not 0 unless the owner unfollow their site

Good point.  Updated to <= 1.  Thanks

> I can't see the grow action card. Is there any other required rule?

The cards appear only once per site and once they are dismissed or action is taken then they won't appear ever again.

> Will we develop this card in v2?

We're doing whatever is on the final designs section J0oHgsu9i7jBQneidWemb5-fi-1721%3A26059 only.  Nothing outside of it, which were all part of iterations before it.

**Comment by AliSoftware** [comment]
Moving the milestone of this PR to `20.1`, as I'm doing the code freeze of `20.0` today.

⚠️  Note that we had to revert the [FluxC PR #2413](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2413) that this PR depends on, because this PR 2413 got merged in FluxC `trunk` too soon, i.e. before this WPAndroid PR 16608 landed.
Having the FluxC PR land before the WPAndroid PR that needs it landed means that:
 - We encountered conflicts when trying to merge the final WPAndroid `release/19.9` into `trunk`
 - We couldn't point the code freeze of WPAndroid to the tag of FluxC that was done during WooCommerce's code freeze, as that tag included all the latest commits from FluxC `trunk`, so including 2413 which landed too soon and shouldn't have been included in the new `1.44.0` FluxC tag in the first place

In the future to avoid creating those conflicts and having us to solve the issue by creating new tags for FluxC and making them diverge from WC, please make sure you only merge FluxC PRs only once the corresponding WPAndroid PR is ready to merge, and merge those together, to avoid FluxC including changes that won't be integrated in WPAndroid in time for the code freeze.

File: WordPress/src/main/res/layout/stats_block_list_action_card.xml
Comment (irfano) [inline WordPress/src/main/res/layout/stats_block_list_action_card.xml:2]: Do we need to use card view here? If I'm not mistaken, this card view is inside another card view (`stats_list_block`) and that causes two-card view strokes on the screen.

**Review by irfano — APPROVED** [review]
LGTM! Great Job! I could see all the action cards. 👍🏻 Check comments ⬇️, then we're good to merge after updating the FluxC reference.
- Check my comment for two nested card views.
- Schedule and reminder cards are not shown at the same time, right? I dismissed the reminder card, then I navigated back and reopened the stats screen, and the schedule card appeared.

File: WordPress/src/main/res/layout/stats_block_list_action_card.xml
Comment (ravishanker) [reply WordPress/src/main/res/layout/stats_block_list_action_card.xml:2]: There's recyclerview inside card view there, where the view holders for each row are assembled.  Yes, it need a card view here since it has a dark background and one static layout unlike other cards.  Removed the corner radius on action card, and updated it on the outer one to 10dp that was due to be done anyways.

**Comment by ravishanker** [comment]
>     * Schedule and reminder cards are not shown at the same time, right? I dismissed the reminder card, then I navigated back and reopened the stats screen, and the schedule card appeared.

correct ✔️

**Pull request merged**
